ui/imports/shared/status/StatusChatInput.qml

```diff
@@ -383,17 +373,44 @@ Rectangle {
         }
     }
 
+    //mentions helper properties
+    property int copyTextStart: 0
+    property string copiedTextPlain: ""
+    property string copiedTextFormatted: ""
+    ListView {
+        id: dummyContactList
+        model: control.usersStore.usersModel
+        delegate: Item {
+            property string contactName: model.name
+        }
+    }
+
     function onRelease(event) {
         if (event.key === Qt.Key_Backspace && textFormatMenu.opened) {
             textFormatMenu.close()
         }
         // the text doesn't get registered to the textarea fast enough
         // we can only get it in the `released` event
+
         if (paste) {
+            if (copiedTextPlain.includes("@")) {
+                copiedTextFormatted = copiedTextFormatted.replace(/underline/g, "none").replace(/span style="/g, "span style=\" text-decoration:none;");
+                for (var j = 0; j < dummyContactList.count; j++) {
+                    var name = dummyContactList.itemAtIndex(j).contactName;
+                    if (copiedTextPlain.indexOf(name) > -1) {
+                        name += '';
+                        copiedTextPlain += '';
+                        var subStr = name.replace(/[.*+?^${}()|[\]\\]/g, '\\$&');
+                        var regex = new RegExp(subStr, 'gi'), result, indices = [];
+                        while ((result = regex.exec(copiedTextPlain))) {
+                            mentionsPos.push({"leftIndex": (result.index + copyTextStart - 1), "rightIndex": (result.index + copyTextStart + name.length)});
+                        }
+                    }
+                }
+            }
+            messageInputField.remove(copyTextStart, (copyTextStart + copiedTextPlain.length));
+            insertInTextInput(copyTextStart, copiedTextFormatted);
```

Comment:
One question/concern I have with this is , what happens if you copy text from the input, then copy text from outside the app (from your browser for example? I think the paste would then put the first copied text, which would confuse the user.

I'm not sure how we can fix that to be honest. Maybe it's a rare occurrence, or also maybe I misunderstood the code and it wouldn't do that anyway?

Reply:
Yeah! When you copy outside the app, it makes a mix of the text outside and the last copied text inside, writing as a firstly the inside one and then concatenate with the outside.

See the video for better understanding!

https://user-images.githubusercontent.com/97019400/156158592-2fa423a9-143d-41e9-88c2-de769e176bed.mov

Reply:
This should be fixed now, please check latest version, have in mind it depends on https://github.com/status-im/dotherside/pull/59 so you might need to remove DOtherSide lib and build again to see the changes.